specs/phase0/fork-choice.md

```diff
@@ -304,13 +366,69 @@ def store_target_checkpoint_state(store: Store, target: Checkpoint) -> None:
 
 ```python
 def update_latest_messages(store: Store, attesting_indices: Sequence[ValidatorIndex], attestation: Attestation) -> None:
-    target = attestation.data.target
+    attestation_slot = attestation.data.slot
     beacon_block_root = attestation.data.beacon_block_root
     for i in attesting_indices:
-        if i not in store.latest_messages or target.epoch > store.latest_messages[i].epoch:
-            store.latest_messages[i] = LatestMessage(epoch=target.epoch, root=beacon_block_root)
+        if i not in store.latest_messages or attestation_slot > store.latest_messages[i].slot:
+            node_key = get_block_slot_key(beacon_block_root, attestation_slot)
+            store.latest_messages[i] = LatestMessage(slot=attestation_slot, root=node_key)
 ```
 
+#### `add_block_slot_node`
+
+```python
+def add_block_slot_node(store: Store, block: BeaconBlock, node_slot: Slot) -> BlockSlotNode:
+    """
+    Adds a ``BlockSlotNode`` corresponding to ``block`` at slot ``node_slot`` in ``store.block_slot_tree`` 
+    and returns the same.
+
+    Algorithm:
+        1. Check if the parent block's node exists at slot ``block.slot - 1``. Add the node if it doesn't exist. 
+        2. Check if this block's node exists at slot ``block.slot``. Add the node if it doesn't exist. 
+        3. If ``node_slot > block.slot``, add virtual block nodes from slot ``block.slot + 1`` till ``node_slot``.
+    """
+    assert block.slot <= node_slot
+
+    # Check if parent block's node exists at block.slot-1
+    # Check for parent block's node only if current block is not genesis/anchor block
+    # NOTE: This only checks if ``block`` is the genesis block. Arbitrary anchor block capability
+    # is not implemented yet.
+    if block.slot > 0:
```

Comment:
```suggestion
    if block.slot > GENESIS_SLOT:
```

Reply:
Using GENESIS_SLOT for now, will shift to anchor block slot when adding that capability